### PR TITLE
Fix boost fPIC management

### PR DESCRIPTION
**Pull request by klimkin**
Specify library name and version:  **boost/1.75.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
> MSBUILD : error MSB4166: Child node "4" exited prematurely.

Again....

You'll need to retrigger CI, close the pr wait 10s and then re-open it 🔁

File: recipes/boost/all/conanfile.py [hunk 1]
```diff
@@ -251,7 +251,27 @@ def config_options(self):
     def _configure_options(self):
         return self._dependencies["configure_options"]
 
+    @property
+    def _fPIC(self):
+        try:
+            return self.options.fPIC
+        except ConanException:
+            return False
```
Comment (prince-chrismc) [inline recipes/boost/all/conanfile.py:259]: It's deleted on Unix shared

```suggestion
            return True
```

File: recipes/boost/all/conanfile.py [hunk 2]
```diff
@@ -251,7 +251,27 @@ def config_options(self):
...
+
+    @property
+    def _shared(self):
+        try:
+            return self.options.shared
+        except ConanException:
+            return False
```
Comment (prince-chrismc) [inline recipes/boost/all/conanfile.py:266]: I am curious if there was any motivation against our usual pattern?

```suggestion
        return self.options.get_safe("shared", False)
```

Comment (klimkin) [reply recipes/boost/all/conanfile.py]: Yes, I missed the pattern. Fixed in a bit different way.

**Review by prince-chrismc — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**